src/components/DayOverview/DayOverviewPdf.tsx

```diff
@@ -0,0 +1,202 @@
+import { faFlag, faSquare, IconDefinition } from "@fortawesome/free-solid-svg-icons";
+
+import { ParcelOfSpecificDateLocation } from "@/components/DayOverview/DayOverview";
+interface DayOverviewerPdfProps {
+    date: Date;
+    location: string;
+    data: ParcelOfSpecificDateLocation[];
+}
+
+interface DayOverviewerCardProps extends DayOverviewerPdfProps {}
+
+interface GeneralCellStyle {
+    borderStyle: "solid" | "dashed" | "dotted" | undefined;
+    borderWidth: number;
+    borderLeftWidth: number;
+    borderTopWidth: number;
+    padding: string;
+}
+
+interface CustomSVGProps {
+    icon: IconDefinition;
+    color: string;
+    fill: boolean;
+}
+
+const generalCellStyle: GeneralCellStyle = {
+    borderStyle: "solid",
+    borderWidth: 1,
+    borderLeftWidth: 0,
+    borderTopWidth: 0,
+    padding: "3px",
+};
+
+const styles = StyleSheet.create({
+    page: {
+        display: "flex",
+        flexDirection: "column",
+        fontSize: "12px",
+    },
+    card: {
+        margin: "0px 40px 20px",
+    },
+    table: {
+        borderStyle: "solid",
+        borderWidth: 1,
+        borderRightWidth: 0,
+        borderBottomWidth: 0,
+    },
+    title: {
+        fontFamily: "Helvetica-Bold",
+        textTransform: "uppercase",
+        fontSize: "30px",
+        textAlign: "center",
+        margin: "30px",
+    },
+    dateLocationWrap: {
+        display: "flex",
+        flexDirection: "row",
+        justifyContent: "space-between",
+        marginBottom: "5px",
+    },
+    svg: {
+        margin: "0 1px",
+        padding: "0 1px",
+    },
+    row: { display: "flex", flexDirection: "row" },
+    bold: { fontFamily: "Helvetica-Bold" },
+
+    // headers
+    header_flag: { flex: 1, fontFamily: "Helvetica-Bold", ...generalCellStyle },
+    header_name: { flex: 4, fontFamily: "Helvetica-Bold", ...generalCellStyle },
+    header_postcode: { flex: 4, fontFamily: "Helvetica-Bold", ...generalCellStyle },
+    header_time: { flex: 2, fontFamily: "Helvetica-Bold", ...generalCellStyle },
+    header_collection: { flex: 4, fontFamily: "Helvetica-Bold", ...generalCellStyle },
+    header_six: { flex: 10, fontFamily: "Helvetica-Bold", ...generalCellStyle },
+
+    // cells
+    cell_logo: { flex: 1, ...generalCellStyle },
+    cell_name: { flex: 4, ...generalCellStyle },
+    cell_postcode: { flex: 4, ...generalCellStyle },
+    cell_time: { flex: 2, ...generalCellStyle },
+    cell_collection: { flex: 4, ...generalCellStyle },
+    cell_instructions: { flex: 10, ...generalCellStyle },
+});
+
+const dateTimeToAMPM = (datetime: string): string => {
+    return new Date(datetime).getHours() <= 11 ? "AM" : "PM";
+};
+
+const CustomSVG: React.FC<CustomSVGProps> = ({ icon, color, fill }) => (
+    <Svg width={12} height={12} viewBox="0 0 512 512" style={styles.svg}>
+        <Path
+            fill={fill ? color : undefined}
+            stroke={color}
+            strokeWidth={40}
+            d={icon.icon[4] as string}
```

Comment:
is there a better way to do this that is not type casting? (not a big issue, just curious)

Reply:
good question, I ain't sure why is `IconDefinition[4]` casted as `string | string[]` at first at all - looked through most of them and don't seem there is any array at all

Reply:
I wonder if changing the interface value could help with this? Not sure (mentioned above as to where)

Reply:
not sure if there needs to be something in place for handling `icon.icon[4]` when it is a `string[]` - maybe even just a ternary / if else - because I doubt they'll put that in the docs if there isn't a use case (although tbf, they might)

I'll leave this for you to decide though :)

Reply:
I have tried to implement some logic to ensure it is definitely string (just in case I overlooked the library)
See if you find it ok or not?